### ISSUE TITLE
Incorrect conversion of string into Number if the number is prefixed with ZEROs

### ISSUE THREAD

**Issue by lgiuliani80** [nilproject/NiL.JS#294]
| Expression | Correct result        | Actual NiL.Js result        |
|-------------|---------------------|---------------------------|
| `+"020"`       | `20`                          | `20`                                  |
| `020`            | `16`                          | `16`                                  |
| `Number("020")` | `20`                   | **`16`**                                  |

In other words octal notation with the "0" prefix before the number shall only be applied for LITERAL numeric values in the JS code and NOT in conversions from strings.
